File: src/13.ts
import _ from 'lodash';
import slurp from './core/slurp';

// Compute the first ten digits of the sum of the one-hundred 50-digit numbers defined by `13.txt`.
//
// See {@link https://projecteuler.net/problem=13}.
export default function compute() {
  const numbers = slurp('13.txt').split('\n').map(BigInt);
  const sum = numbers.reduce((a, b) => a + b, 0n);
  const text = sum.toString().slice(0, 10);
  return _.toNumber(text);
}

File: project_euler/13.py
# Problem 13: Large Sum
#
# Work out the first ten digits of the sum of the one-hundred 50 digit numbers.
#
# See https://projecteuler.net/problem=13
from pathlib import Path


def run():
    file = Path(__file__).parent / "data" / "13.txt"
    xs = [int(x) for x in file.read_text().splitlines()]
    value = str(sum(xs))
    return int(value[:10])


def test_run():
    assert run() == 5537376230
